broker/src/main/java/org/apache/rocketmq/broker/BrokerStartup.java

```diff
@@ -187,6 +187,7 @@ public static BrokerController createBrokerController(String[] args) {
             JoranConfigurator configurator = new JoranConfigurator();
             configurator.setContext(lc);
             lc.reset();
+            System.setProperty("brokerLogDir", brokerConfig.getBrokerName() + "_" + brokerConfig.getBrokerId());
```

Comment:
In dleger mode, The broker id changes when the leader change. so if we restart the leader you will see broker id change from zero to non-zero

Reply:
i 

> In dleger mode, The broker id changes when the leader change. so if we restart the leader you will see broker id change from zero to non-zero

I think brokerport can be used

Reply:
> i
> 
> > In dleger mode, The broker id changes when the leader change. so if we restart the leader you will see broker id change from zero to non-zero
> 
> I think brokerport can be used

If brokers are deployed in Dledger mode, use dLegerSelfId instead of brokerId